### PR TITLE
PSRV-438 Fix mysql data dir perms

### PR DESCRIPTION
**Pull request by mreeves1**

### CODE REVIEW

**Comment by mreeves1** [comment]
@mkantor , @BrianHoltz take a look. It's a bit ghetto but I think the right approach given the context.

slack convo describing the issue:
https://rightscalecommunity.slack.com/archives/kabam-ms/p1472169088000413

Basically because the directory is owned by the mysql user but the contents are owned by uid of 106 we need to explicitly change ownership to mysql because snapshot host's uid mapping may differ from  recipient.

**Comment by BrianHoltz** [comment]
Your chown call is totally UNIX/POSIX compliant, so if that's ghetto then I'm straight outta that hood. :-)

http://pubs.opengroup.org/onlinepubs/9699919799/utilities/chown.html

File: recipes/volume.rb [hunk 1]
```diff
@@ -103,6 +103,13 @@
   action :create
 end
 
+# We may inherit the volume from another machine's backup
+# The userids may differ so we must set them explicitly to mysql
+execute 'explicitly set mysql data directory recursive file ownership' do
+  command "chown -R mysql:mysql #{new_mysql_dir}"
```
Comment (mkantor) [inline recipes/volume.rb]: Any possibility of spaces in the path? You may want to quote it: `command "chown -R mysql:mysql '#{new_mysql_dir}'"`

**Comment by mkantor** [comment]
Makes sense.

File: recipes/volume.rb (hunk 1 above)
Comment (mreeves1) [inline recipes/volume.rb]: done.

**Pull request merged**